src/service/artifacts/artifact-service.ts

```diff
@@ -9,32 +8,32 @@ import { UploadResponse } from "@actions/artifact";
 
 @Service()
 export class ArtifactService {
-  private readonly configService: ConfigurationService;
   private readonly uploadService: UploadService;
   private readonly logger: LoggerService;
 
   constructor() {
-    this.configService = Container.get(ConfigurationService);
     this.uploadService = Container.get(UploadService);
     this.logger = LoggerServiceFactory.getInstance();
   }
 
-  private getNodesToArchive(nodeChain: Node[]): Node[] {
-    const startingProject = this.configService.getStarterNode();
-    const dependencies = startingProject.archiveArtifacts?.dependencies ?? ArchiveDependencies.NONE;
+  private getNodesToArchive(nodeChain: Node[], startingNode: Node): Node[] {
+    const dependencies = startingNode.archiveArtifacts?.dependencies ?? ArchiveDependencies.NONE;
     let result: Node[];
-    if (dependencies === ArchiveDependencies.NONE) {
-      result = [startingProject];
-    } else if (dependencies === ArchiveDependencies.ALL) {
-      result = nodeChain.filter((node) => !!node.archiveArtifacts);
-    } else {
-      result = nodeChain.filter((node) => node.archiveArtifacts && (dependencies.includes(node.project) || this.configService.isNodeStarter(node)));
+    switch (dependencies) {
+      case ArchiveDependencies.NONE:
+        result = [startingNode];
+        break;
+      case ArchiveDependencies.ALL:
+        result = nodeChain.filter((node) => !!node.archiveArtifacts);
```

Comment:
is that parenthesis added by lint or something? :thinking: 
```suggestion
        result = nodeChain.filter(node => !!node.archiveArtifacts);
```